dao/StudentDao.java

```diff
@@ -39,7 +47,28 @@ public void update(Student newValues) {
     }
 
     @Override
-    public void delete(String studentId) {
+    public void delete(String studentId)  {
         students.remove(get(studentId));
     }
+
+    @Override
+    public List<Student> sortStudentsById() {
+        return new ArrayList<>(
```

Comment:
Here you could even pass the `Comparator<Student>` as an argument to let the user of the method decide how she or he wants to sort the list of students.

```java
public List<Student> sortStudents(Comparator<Student> by) {
    return students.stream()
        .sorted(by)
        .toList();
}
...
sortStudents(Comparator.comparing(Student::getStudentId));
```